lib/RestructureCFG/InlineDivergentScopesPass.cpp

```diff
@@ -279,6 +279,49 @@ electDivergence(BasicBlock *Candidate,
   return std::nullopt;
 }
 
+/// Helper function which create a single entry point block where a divergence
+/// is entered into by multiple successors of a `Conditional`
+static void createHead(BasicBlock *Conditional,
+                       DivergenceDescriptor &Divergence,
+                       BasicBlock *PlaceHolderTarget) {
+  LLVMContext &Context = getContext(Conditional);
+  Function *F = Conditional->getParent();
+  Instruction *ConditionalTerminator = Conditional->getTerminator();
+  BasicBlock *Head = BasicBlock::Create(Context,
+                                        Conditional->getName() + "_head_ids",
+                                        F);
+  Instruction *HeadTerminator = ConditionalTerminator->clone();
+  IRBuilder<> HeadBuilder(Head);
+  HeadBuilder.Insert(HeadTerminator);
+
+  // Collect the `Successor`s composing the local `Divergence`
+  SmallSet<BasicBlock *, 4> LocalDivergentSuccessors;
+  LocalDivergentSuccessors.insert(Divergence.DivergentSuccessors.begin(),
+                                  Divergence.DivergentSuccessors.end());
+
+  // And collect all the `Successor`s that do not make up the local
+  // `Divergence`
+  SmallVector<BasicBlock *> LocalNonDivergentSuccessors;
+  for (BasicBlock *Successor : children<Scope<BasicBlock *>>(Head)) {
+    if (not LocalDivergentSuccessors.contains(Successor)) {
+      LocalNonDivergentSuccessors.push_back(Successor);
+    }
+  }
+
+  // The `Head` only connects the `LocalDivergentSuccessors`, we simplify
+  // away the other successors
+  for (BasicBlock *Successor : LocalNonDivergentSuccessors) {
+    HeadTerminator->replaceSuccessorWith(Successor, PlaceHolderTarget);
+  }
+  simplifyTerminator(Head, PlaceHolderTarget);
```

Comment:
The only other use of `simplifyTerminator` follows this same pattern: first you replace successors with placeholder target, then you call `simplifyTerminator`. Please refactor.

Reply:
Since the `for` and `replaceSuccessorWith` pattern was used multiple times in the file, I crafted a new `replaceSuccessors` helper here https://github.com/revng/revng/blob/3a9379ef740d1d2b9f7bcad02cc962719251e02a/lib/RestructureCFG/InlineDivergentScopesPass.cpp#L333C3-L333C20 and substituted all the uses